modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRow.es.js

```diff
@@ -0,0 +1,326 @@
+ */
+function canDrop(props, monitor) {
+	const {groupId: destGroupId, index: destIndex} = props;
+	const {groupId: startGroupId, index: startIndex} = monitor.getItem();
+
+	return destGroupId !== startGroupId || destIndex !== startIndex;
+}
+
+/**
+ * Implements the behavior of what will occur when an item is dropped.
+ * Items dropped on top of rows will create a new grouping.
+ * This method must be called `drop`.
+ * @param {Object} props Component's current props.
+ * @param {DropTargetMonitor} monitor
+ */
+function drop(props, monitor) {
+	const {
+		criterion,
+		groupId: destGroupId,
+		index: destIndex,
+		onChange,
+		onMove,
+		supportedOperators
+	} = props;
+
+	const {
+		criterion: droppedCriterion,
+		groupId: startGroupId,
+		index: startIndex
+	} = monitor.getItem();
+
+	const {operatorName, propertyName, value} = droppedCriterion;
+
+	const newCriterion = {
+		operatorName: operatorName ?
+			operatorName :
+			supportedOperators[0].name,
+		propertyName,
+		value: value ? value : ''
+	};
+
+	const newGroup = {
+		conjunctionName: CONJUNCTIONS.AND,
+		groupId: generateGroupId(),
+		items: [criterion, newCriterion]
+	};
+
+	const itemType = monitor.getItemType();
+
+	if (itemType === DragTypes.PROPERTY) {
+		onChange(newGroup);
+	}
+	else if (itemType === DragTypes.CRITERIA_ROW) {
+		onMove(
+			startGroupId,
+			startIndex,
+			destGroupId,
+			destIndex,
+			newGroup,
+			true
+		);
+	}
+}
+
+/**
+ * Passes the required values to the drop target.
+ * This method must be called `beginDrag`.
+ * @param {Object} props Component's current props
+ * @returns {Object} The props to be passed to the drop target.
+ */
+function beginDrag({criterion, groupId, index}) {
+	return {criterion, groupId, index};
+}
+
+class CriteriaRow extends Component {
+	static propTypes = {
+		canDrop: PropTypes.bool,
+		connectDragPreview: PropTypes.func,
+		connectDragSource: PropTypes.func,
+		connectDropTarget: PropTypes.func,
+		criterion: PropTypes.object,
+		dragging: PropTypes.bool,
+		editing: PropTypes.bool,
+		groupId: PropTypes.string,
+		hover: PropTypes.bool,
+		index: PropTypes.number,
+		modelLabel: PropTypes.string,
+		onAdd: PropTypes.func,
+		onChange: PropTypes.func,
+		onDelete: PropTypes.func,
+		onMove: PropTypes.func,
+		supportedConjunctions: PropTypes.array,
+		supportedOperators: PropTypes.array,
+		supportedProperties: PropTypes.array,
+		supportedPropertyTypes: PropTypes.object
+	};
+
+	static defaultProps = {
+		editing: true
+	};
+
+	_getReadableCriteriaString(modelLabel, propertyLabel, operatorLabel, value) {
+		return sub(
+			Liferay.Language.get('x-with-property-x-x-x'),
+			[
+				<span key="model-name">
+					{modelLabel}
+				</span>,
+				<b key="property">
+					{propertyLabel}
+				</b>,
+				<span key="operator">
+					{operatorLabel}
+				</span>,
+				<b key="value">
+					{value}
+				</b>
+			],
+			false
+		);
+	}
+
+	_getSelectedItem = (list, idSelected) =>
```

Comment:
For these implicit returns, how come sometimes you are forced to wrap it with brackets and other times not? Is that just a nuance with csf? I noticed you'll sometimes use implicit return and other times its `{return fooBar;}`

Reply:
For what its worth, I prefer the implicit returns but I remember you saying its something to do with csf

Reply:
It's only forced to wrap in brackets if the implicit return is a ternary like `=> isTrue ? 'yes' : 'no'`.